docs/table-partitioning-bucketing-sort-order.md

```diff
@@ -0,0 +1,294 @@
+**Not all catalogs may support those features.**. Please refer to the related document for more details.
+:::
+
+The following is an example of creating a partitioned, bucketed table, and sorted order table:
+
+<Tabs>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X POST -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+  "name": "table",
+  "columns": [
+    {
+      "name": "id",
+      "type": "integer",
+      "nullable": true,
+      "comment": "Id of the user"
+    },
+    {
+      "name": "name",
+      "type": "varchar(2000)",
+      "nullable": true,
+      "comment": "Name of the user"
+    },
+    {
+      "name": "age",
+      "type": "short",
+      "nullable": true,
+      "comment": "Age of the user"
+    },
+    {
+      "name": "score",
+      "type": "double",
+      "nullable": true,
+      "comment": "Score of the user"
+    }
+  ],
+  "comment": "Create a new Table",
+  "properties": {
+    "format": "ORC"
+  },
+  "partitioning": [
+    {
+      "strategy": "identity",
+      "fieldName": ["score"]
+    }
+  ],
+  "distribution": {
+    "strategy": "hash",
+    "number": 4,
+    "funcArgs": [
+      {
+        "type": "field",
+        "fieldName": ["score"]
+      }
+    ]
+  },
+  "sortOrders": [
+    {
+      "direction": "asc",
+      "nullOrder": "NULLS_LAST",
+      "sortTerm":  {
+        "type": "field",
+        "fieldName": ["name"]
+      }
+    }
+  ]
+}' http://localhost:8090/api/metalakes/metalake/catalogs/catalog/schemas/schema/tables
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+tableCatalog.createTable(
+    NameIdentifier.of("metalake", "hive_catalog", "schema", "table"),
+    new ColumnDTO[] {
+      ColumnDTO.builder()
+        .withComment("Id of the user")
+        .withName("id")
+        .withDataType(Types.IntegerType.get())
+        .withNullable(true)
+      .build(),
+      ColumnDTO.builder()
+        .withComment("Name of the user")
+        .withName("name")
+        .withDataType(Types.VarCharType.of(1000))
+        .withNullable(true)
+        .build(),
+      ColumnDTO.builder()
+        .withComment("Age of the user")
+        .withName("age")
+        .withDataType(Types.ShortType.get())
+        .withNullable(true)
+        .build(),
+      ColumnDTO.builder()
+        .withComment("Score of the user")
+        .withName("score")
+        .withDataType(Types.DoubleType.get())
+        .withNullable(true)
+        .build(),
+    },
+    "Create a new Table",
+    tablePropertiesMap,
+    new Transform[] {
+    // Partition by id
+      Transforms.identity("score")
+    },
+    // CLUSTERED BY id
+    new DistributionDTO.Builder()
+      .withStrategy(Strategy.HASH)
+      .withNumber(4)
+      .withArgs(FieldReferenceDTO.of("id"))
+      .build(),
+    // SORTED BY name asc
+    new SortOrderDTO[] {
+      SortOrders.of(FieldReferenceDTO.of("score"), SortDirection.ASCENDING, NullOrdering.NULLS_LAST)
+    });
```

Comment:
I think we should not use xxxDTO, instead we should use some methods in APIs, right? @mchades

Reply:
Yes, there are several static methods in APIs, `xxxDTO` is a representation of the intermediate result of JSON serialization.

Reply:
The method isn't yet available in the APIs, 
<img width="1440" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/b80f81bc-3312-43c1-bea5-fa8b0da42dba">

So, do I need to add it to this PR or use another PR later?

Reply:
use `com.datastrato.gravitino.rel.expressions.NamedReference#field(java.lang.String)`

Reply:
Please fix it here.

Reply:
There are two follow-things:

1. "Column" should have an implementation used by client, not DTO, @mchades can you please work on this?
2. Make sure all the IT codes use APIs, not DTOs for partitioning/sorting/distribution. @yuqi1129 can you please work on this?

Reply:
> Please fix it here.

done

> Please fix it here.

done

Reply:
> 1. "Column" should have an implementation used by client, not DTO

Tracked by #1292